jujugui/static/gui/src/app/state/test-state.js

```diff
@@ -0,0 +1,566 @@
+        state: { gui: { applications: '', inspector: 'ghost' } }
+      }, {
+        parts: ['inspector', 'service123', 'relate-to', 'serviceabc'],
+        state: { gui: { inspector: 'service123/relate-to/serviceabc' } }
+      }, {
+        parts: [
+          'inspector', 'apache2', 'machines', '3', 'lxc-0', 'deploy', 'foo'],
+        state: {
+          gui: { inspector: 'apache2', machines: '3/lxc-0', deploy: 'foo' }
+        }
+      }];
+
+      guiSections.forEach(section => {
+        assert.deepEqual(
+          state._parseGUI(section.parts, {}),
+          section.state,
+          `${section.path} did not properly generate the state object: ` +
+          JSON.stringify(section.state));
+      });
+    });
+  });
+
+  describe('State._parseUser()', () => {
+    it('populates the user and store portions of the state object', () => {
+      const state = new window.jujugui.State({
+        baseURL: 'http://abc.com:123'
+      });
+
+      const userSections = [{
+        parts: [],
+        state: {},
+        outputParts: []
+      }, {
+        parts: ['u', 'ant'],
+        state: {
+          user: 'ant'
+        },
+        outputParts: []
+      }, {
+        parts: ['u', 'frankban', 'settings'],
+        state: {
+          user: 'frankban/settings'
+        },
+        outputParts: []
+      }, {
+        parts: ['u', 'hatch', 'staging', 'haproxy'],
+        state: {
+          user: 'hatch/staging'
+        },
+        outputParts: ['haproxy']
+      }, {
+        parts: ['u', 'frankban', 'production', 'ghost', 'xenial'],
+        state: {
+          user: 'frankban/production'
+        },
+        outputParts: ['ghost', 'xenial']
+      }, {
+        parts: ['u', 'hatch', 'staging', 'ghost', '42'],
+        state: {
+          user: 'hatch/staging'
+        },
+        outputParts: ['ghost', '42']
+      }, {
+        parts: ['u', 'hatch', 'mongodb', 'xenial'],
+        state: {
+          store: 'hatch/mongodb/xenial'
+        },
+        outputParts: []
+      }, {
+        parts: ['u', 'hatch', 'mongodb', '47'],
+        state: {
+          store: 'hatch/mongodb/47'
+        },
+        outputParts: []
+      }, {
+        parts: ['u', 'frankban', 'django', 'bundle', '0'],
+        state: {
+          store: 'frankban/django/bundle/0'
+        },
+        outputParts: []
+      }, {
+      // Multi user delimiter handling.
+        parts: ['u', 'hatch', 'staging', 'u', 'frankban', 'django'],
+        state: {
+          user: 'hatch/staging',
+          store: 'frankban/django'
+        },
+        outputParts: []
+      }, {
+        parts: [
+          'u', 'frankban', 'production', 'u', 'hatch', 'mongodb', 'xenial'],
+        state: {
+          user: 'frankban/production',
+          store: 'hatch/mongodb/xenial'
+        },
+        outputParts: []
+      }, {
+        parts: [
+          'u', 'hatch', 'staging', 'u', 'hatch', 'django', 'bundle', '0'],
+        state: {
+          user: 'hatch/staging',
```

Comment:
So teh user state stores both the profile pages and the model? I guess we could easily separate the two cases already?